cluster/cluster.go

```diff
@@ -707,6 +708,16 @@ func ParseConfig(clusterFile string) (*v3.RancherKubernetesEngineConfig, error)
 	if err != nil {
 		return nil, err
 	}
+
+	// Calico FelixConfigurationSpec is a 3rd party type that doesn't unmarshall properly
+	// so we need to parse it manually and inject the parsed struct
+	// into the final rkeConfig
+	var felixConfig calico.FelixConfigurationSpec
+	clusterFile, err = resolveFelixConfiguration(clusterFile, &felixConfig)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I would prefer this to be hidden in calico.go and only execute it if `if rkeConfig.Network.Plugin == "calico" {`, that way we don't need to import calico in `cluster.go`. This is personal taste :)